apps/web/src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -3,40 +3,37 @@ import buildConversationId from '@lib/buildConversationId';
 import chunkArray from '@lib/chunkArray';
 import { buildConversationKey, parseConversationKey } from '@lib/conversationKey';
 import conversationMatchesProfile from '@lib/conversationMatchesProfile';
-import type { Conversation, Stream } from '@xmtp/xmtp-js';
+import type { Conversation } from '@xmtp/xmtp-js';
 import { SortDirection } from '@xmtp/xmtp-js';
 import type { DecodedMessage } from '@xmtp/xmtp-js/dist/types/src/Message';
 import type { Profile } from 'lens';
 import { useProfilesLazyQuery } from 'lens';
-import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 
 const MAX_PROFILES_PER_REQUEST = 50;
 
-const useMessagePreviews = () => {
-  const router = useRouter();
+const useMessagePreviews = (currentIndex: number) => {
   const currentProfile = useAppStore((state) => state.currentProfile);
   const conversations = useMessageStore((state) => state.conversations);
-  const setConversations = useMessageStore((state) => state.setConversations);
   const messageProfiles = useMessageStore((state) => state.messageProfiles);
   const setMessageProfiles = useMessageStore((state) => state.setMessageProfiles);
   const previewMessages = useMessageStore((state) => state.previewMessages);
   const setPreviewMessages = useMessageStore((state) => state.setPreviewMessages);
   const selectedProfileId = useMessageStore((state) => state.selectedProfileId);
-  const setSelectedProfileId = useMessageStore((state) => state.setSelectedProfileId);
   const setPreviewMessage = useMessageStore((state) => state.setPreviewMessage);
-  const reset = useMessageStore((state) => state.reset);
-  const { client, loading: creatingXmtpClient } = useXmtpClient();
+  const { client } = useXmtpClient();
```

Comment:
We only want to call `useXmtpClient()` once to avoid multiple signature reqs from multiple hooks. Since the loading state is now managed in `useListConversations`, this can be:

```suggestion
  const { client } = useMessageStore((state) => state.client);
```